packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -23,21 +23,24 @@ class VMService {
   }
 
   /// Connect to '127.0.0.1' at [port].
-  static Future<VMService> connect(int port) async {
-    Uri uri = new Uri(scheme: 'ws', host: '127.0.0.1', port: port, path: 'ws');
+  static Future<VMService> connect(Uri httpUri) async {
+    String wsPath = httpUri.path;
+    if (!wsPath.endsWith('/'))
```

Comment:
Also a cleaner way of dealing with paths is to use the path package's join method.